pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/segment_generation_and_push/SegmentGenerationAndPushTaskExecutor.java

```diff
@@ -221,29 +222,39 @@ private URI moveSegmentToOutputPinotFS(Map<String, String> taskConfigs, File loc
     return outputSegmentTarURI;
   }
 
-  private PinotFS getInputPinotFS(Map<String, String> taskConfigs, URI fileURI) {
+  private PinotFS getInputPinotFS(Map<String, String> taskConfigs, URI fileURI)
+      throws Exception {
     String fileURIScheme = fileURI.getScheme();
     if (fileURIScheme == null) {
-      fileURIScheme = PinotFSFactory.LOCAL_PINOT_FS_SCHEME;
+      return LOCAL_PINOT_FS;
     }
-    if (!PinotFSFactory.isSchemeSupported(fileURIScheme)) {
-      String fsClass = taskConfigs.get(BatchConfigProperties.INPUT_FS_CLASS);
+    // Try to create PinotFS using given Input FileSystem config always
+    String fsClass = taskConfigs.get(BatchConfigProperties.INPUT_FS_CLASS);
```

Comment:
The batch configs won't be changed in the middle when the minion servers are running. So how do the other minion tasks create new S3PinotFS with their own AWS S3 credentials? Does that mean there could be two places that store the credentials?

Reply:
In this case, users can pass AWS credentials as part of task spec to use a different was credential.
Each minion task should use its own S3 credential to init PinotFS unless it's not there, then we fall back to use the PinotFS created during minion starts.

Reply:
You can always assume that there are two AWS credentials:
1. Pinot Credential: this is used by pinot only to leverage s3 as the deep store.
2. Data ingestion Credential: this is per user basis, it requires another AWS credential for data reading.